tests/paramgrill.c

```diff
@@ -140,6 +179,126 @@ static unsigned longCommandWArg(const char** stringPtr, const char* longCommand)
+    U32 dSpeed;
+    U32 cMem;    /* bytes */    
+} constraint_t;
+
+#define CLAMPCHECK(val,min,max) {                     \
+    if (val && (((val)<(min)) | ((val)>(max)))) {     \
+        DISPLAY("INVALID PARAMETER CONSTRAINTS\n");   \
+        return 0;                                     \
+}   }
+
+
+/* Like ZSTD_checkCParams() but allows 0's */
+/* no check on targetLen? */
+static int cParamValid(ZSTD_compressionParameters paramTarget) {
+    CLAMPCHECK(paramTarget.hashLog, ZSTD_HASHLOG_MIN, ZSTD_HASHLOG_MAX);
+    CLAMPCHECK(paramTarget.searchLog, ZSTD_SEARCHLOG_MIN, ZSTD_SEARCHLOG_MAX);
+    CLAMPCHECK(paramTarget.searchLength, ZSTD_SEARCHLENGTH_MIN, ZSTD_SEARCHLENGTH_MAX);
+    CLAMPCHECK(paramTarget.windowLog, ZSTD_WINDOWLOG_MIN, ZSTD_WINDOWLOG_MAX);
+    CLAMPCHECK(paramTarget.chainLog, ZSTD_CHAINLOG_MIN, ZSTD_CHAINLOG_MAX);
+    if(paramTarget.targetLength > ZSTD_TARGETLENGTH_MAX) {
+        DISPLAY("INVALID PARAMETER CONSTRAINTS\n");
+        return 0;
+    }
+    if(paramTarget.strategy > ZSTD_btultra) {
+        DISPLAY("INVALID PARAMETER CONSTRAINTS\n");
+        return 0;
+    }
+    return 1;
+}
+
+static void cParamZeroMin(ZSTD_compressionParameters* paramTarget) {
+    paramTarget->windowLog = paramTarget->windowLog ? paramTarget->windowLog : ZSTD_WINDOWLOG_MIN;
+    paramTarget->searchLog = paramTarget->searchLog ? paramTarget->searchLog : ZSTD_SEARCHLOG_MIN;
+    paramTarget->chainLog = paramTarget->chainLog ? paramTarget->chainLog : ZSTD_CHAINLOG_MIN;
+    paramTarget->hashLog = paramTarget->hashLog ? paramTarget->hashLog : ZSTD_HASHLOG_MIN;
+    paramTarget->searchLength = paramTarget->searchLength ? paramTarget->searchLength : ZSTD_SEARCHLENGTH_MIN;
+    paramTarget->targetLength = paramTarget->targetLength ? paramTarget->targetLength : 0;
+}
+
+static void BMK_translateAdvancedParams(const ZSTD_compressionParameters params)
+{
+    DISPLAY("--zstd=windowLog=%u,chainLog=%u,hashLog=%u,searchLog=%u,searchLength=%u,targetLength=%u,strategy=%u \n",
+             params.windowLog, params.chainLog, params.hashLog, params.searchLog, params.searchLength, params.targetLength, (U32)(params.strategy));
+}
+
+/* checks results are feasible */
+static int feasible(const BMK_result_t results, const constraint_t target) {
+    return (results.cSpeed >= target.cSpeed) && (results.dSpeed >= target.dSpeed) && (results.cMem <= target.cMem || !target.cMem);
+}
+
+#define EPSILON 0.001
+static int epsilonEqual(const double c1, const double c2) {
+    return MAX(c1/c2,c2/c1) < 1 + EPSILON;
+}
+
+/* checks exact equivalence to 0, to stop compiler complaining fpeq */
+static int eqZero(const double c1) {
+    const double z1 = 0.0;
+    const double z2 = -0.0;
+    return !(memcmp(&c1, &z1, sizeof(double))) || !(memcmp(&c1, &z2, sizeof(double)));
+}
+
+/* returns 1 if result2 is strictly 'better' than result1 */
+/* strict comparison / cutoff based */
+static int objective_lt(const BMK_result_t result1, const BMK_result_t result2) {
```

Comment:
Here also, I'm surprised by the name of the function.
The associated comment is clear enough though.
Try to find a name that reflects the intent.